test/phoenix_component/declarative_assigns_test.exs

```diff
@@ -1310,6 +1310,28 @@ defmodule Phoenix.ComponentDeclarativeAssignsTest do
     end
   end
 
+  test "does not raise if attr :values is not a list, but still an enum" do
+    defmodule Phoenix.ComponentTest.AttrValueTypeMismatch do
+      use Elixir.Phoenix.Component
+
+      attr :foo, :integer, values: 1..10
+      def func(assigns), do: ~H[]
+    end
+  end
+
+  test "raise if attr :default is not in range" do
+    msg = ~r'expected the default value for attr :foo to be one of 1\.\.10, got: 11'
+
+    assert_raise CompileError, msg, fn ->
+      defmodule Phoenix.ComponentTest.AttrDefaultValuesMismatch do
+        use Elixir.Phoenix.Component
+
+        attr :foo, :integer, default: 11, values: 1..10
+        def func(assigns), do: ~H[]
+      end
+    end
+  end
```

Comment:
I think we should also add tests that the validation happens as desired?

Reply:
you mean a test with a default value within the values range?

Reply:
That a value within the range does not warn and a value outside the range warns. :)

Reply:
got it, it's `verify_test.exs` I should amend

Reply:
Done. ✅ 
Also improved tests to count how many warnings were emitted and ensure that good values did not raise